lib/chef/knife/core/subcommand_loader.rb

```diff
@@ -23,7 +23,7 @@ class Knife
     class SubcommandLoader
 
       MATCHES_CHEF_GEM = %r{/chef-[\d]+\.[\d]+\.[\d]+}.freeze
-      MATCHES_THIS_CHEF_GEM = %r{/chef-#{Chef::VERSION}/}.freeze
+      MATCHES_THIS_CHEF_GEM = %r{/chef-#{Chef::VERSION}(-x86-mingw32)?/}.freeze
```

Comment:
@jdmundrawala on a second thought: not sure what to actually match here, i.e. no clue how other valid platforms the chef gem will be built for in the future would look like.

Is it "x86-mingw64" or "i386-mingw64" or just "mingw64" etc?

Maybe it would be better to have an integration test which ensures that `knife cookbook --help` returns the expected subcommands instead?